content/docs/philosophy/goals.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Goals and Non-Goals
+sort_rank: 3
+---
+
+# Goals
+
+## Resilience
+
+First and foremost, Prometheus must be resilient in operation.
+
+
+## Reliable alerting
+
+As a monitoring system, Prometheus is being relied upon to alert humans that
+they need to take action in order to prevent undesired system state.
+
+Thus, its most important function is to keep the pipeline of ingestion, rule
+evaluation, and alert hand-off working.
+
+The second most important function is to give humans context about these alerts
+by allowing access to the most recent data Prometheus ingested
+
+### Resulting design decisions and patterns
+
+Note that this goal might result in widely different design decisions and thus
+operational patterns for different parts of our ecosystem:
+
+For Prometheus itself, this means running every instance as an island of data
+completely detached from every other instance, except for optional federation.
+
+For Alertmanager on the other hand, it means the exact opposite: meshing all
+instances closely together, sharing knowledge about alerts and their
+notifications.
+
+## Simple operation
+
+Operation of Prometheus should be as easy and failure-tolerant as possible. We
+try to put required complexity into earlier phases, going through them less
+often and ideally still while under the control of a smaller subset of people.
+
+One example of this would be the preference of statically linked binaries over
+dynamically-built ones.
+
+## Automation
+
+Computers are good at doing the same thing over and over again, and quickly.
+Humans tend to be better at creative tasks.
+
+Prometheus will always strive to automate away all tasks whenever possible
+through various means; some specific implementations would be service discovery,
+label rewriting, and alert generation.
+
+
+# Non-Goals
+
+# Event handling
+
+Prometheus is dealing with metrics. As such, it will never process and store
+events.
+
+The only exception in our ecosystem is Alertmanager which deals with individual
+alerts and alert groups.
+
+For ways to deal with events, see TODO patterns.
+
+# Push-type system
+
+Prometheus is, and always will be, a pull-type system. We strongly believe that
+this makes operational sense in all but the very largest of scales.
```

Comment:
This is incorrect, we believe it works at all scales.

Also both ways work at scale. The more salient point is that a lot of things we do only really work with push.

Reply:
I would argue that Monarch-style mixed systems allow for even more scale. This is beyond this document's scope, though.

Also, I think you meant pull. It makes sense to expand on those things, though. will put in a TODO into patterns and link back to there.

Reply:
I disagree, both can be scaled indefinitely. Also Monarch is more pull than push.

Reply:
Yeah, there's no limits to scale with either approach, but there's the usual known pros+cons for both approaches, and we're quite married to the pull approach, as Prometheus is so designed around being in control of pulling and processing data when *it* wants.

Reply:
I'd just say the first sentence and then link to a blog post (Julius' from a while back perhaps or the FAQ entry. To the vast majority of users this is a non-issue.

Reply:
Yet many users continue to try and make it do push (and tend to be quite unhappy when told their approach won't work and/or be supported), so I think it's worth a few sentences.

Reply:
Changed.